[Alerting V2][Serverless & 9.5][M2] Add get started pages for the experimental alerting system (#6522)
elastic/docs-content: explore-analyze/alerting/kibana-alerting-experimental/get-started/setup.md

{
  "changes": {
    "alerting:v2:enabled": true
  }
}
```

:::{note}
The `/internal/kibana/global_settings` endpoint is an internal API and might change without notice. There is currently no public equivalent.
:::
:::
::::

### Confirm the UI is accessible [alerting-setup-confirm]

After turning on the setting, verify the {{alerting-v2-system}} is accessible in your space:

1. Use the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md) and enter `Alerting v2 preview`.
2. Select the menu item from the results.

If the menu item doesn't appear immediately, refresh the page and search again. It might take a moment for the UI to reflect the updated setting.

## Turn off the system [alerting-setup-turn-off]

To turn off the {{alerting-v2-system}}, set `alerting:v2:enabled` to `false`.

::::{applies-switch}
:::{applies-item} stack: experimental 9.5+

Go to the **Advanced Settings** page and toggle off **alerting:v2:enabled**.
:::

:::{applies-item} serverless: experimental

Use Dev Tools to call the global settings API:

```json
POST kbn:/internal/kibana/global_settings
{
  "changes": {
    "alerting:v2:enabled": false
  }
}
```
:::
::::

> When the setting is off, rule and action policy execution stops, the APIs and UI are hidden, and existing rules and action policies are paused.

Reviewer: Rule engine does not stop running when the setting is off. per https://elastic.slack.com/archives/C0A7DBQBQ9J/p1783342251497669

Author: Fixed in [`8a8f120` (this PR)](https://github.com/elastic/docs-content/pull/6522/changes/8a8f120311643d531fb84ca2b907561b5cd4720a).

> Turning the setting back on resumes execution. Turning it off does not delete any data. Your rules and action policies remain as {{kib}} saved objects, and existing documents in `.rule-events` and `.alert-actions` are preserved.

Reviewer: Turning the setting back on exposes the UI.

Author: Fixed in [`8a8f120` (this PR)](https://github.com/elastic/docs-content/pull/6522/changes/8a8f120311643d531fb84ca2b907561b5cd4720a).

## Next steps

After turning on the system, configure role access so your team can use it:

> - **[Configure access](configure-access.md):** Create or update a role with access to the {{alerting-v2-system}} features and the data streams they write to. Users need at minimum **Rules: All** to create rules. Granting **Alerts: Read** gives a role Kibana triage access and automatic Elasticsearch `read` access to `.rule-events` and `.alert-actions`, with no separate index privilege needed.

Reviewer 2: Seeing that this unordered list consists of only one item, should it still be a list?

Author: Nope, I'll fix. Thanks!

Author: Fixed in [`9f86278` (this PR)](https://github.com/elastic/docs-content/pull/6522/commits/9f862789f156c1e9b34857afbad7b2f82333f177).

- **A space selected**: Rules, action policies, and the privileges that control them are all space-scoped. Decide which space you'll work in before setting things up. Refer to [Manage spaces](/deploy-manage/manage-spaces.md) to create or switch spaces.
- **Connectors configured** (required for notifications): Action policies send notifications through workflows, which require at least one [connector](/deploy-manage/manage-connectors.md), for example, Slack, email, or PagerDuty.
- **Enterprise license** (Stack deployments only, required for notifications): Workflows-based notifications require an Enterprise license. Rules and alert episodes work on any tier. Refer to the subscription page for [Elastic Cloud](https://www.elastic.co/subscriptions/cloud) and [Elastic Stack/self-managed](https://www.elastic.co/subscriptions) for the breakdown of available features and their associated subscription tiers. {applies_to}`stack: ga 9.5+`

## Turn on the system [alerting-setup-turn-on]

> The {{alerting-v2-system}} is controlled by the `alerting:v2:enabled` advanced setting in {{kib}}. This setting is off by default. Turn it on to make the UIs for {{alerting-v2-system}} features available in your space.

Reviewer 3: Aside from the UIs, the APIs will also be available.

Author: How would users know which endpoints are public and how to use them? If there's public docs available, I can mention that APIs are available too and reference those docs.

Reviewer 3: Good point. They cannot be aware unless they check the code. We are in the process of generating the OAS docs, and they will be released as experimental pretty soon. We can add it later.

::::{applies-switch}
:::{applies-item} stack: experimental 9.5+

**Role requirements**

> You must have the `kibana_admin` role or equivalent {{stack-manage-app}} access to turn on this setting.

Author: ```suggestion
You must have the `kibana_admin` role or equivalent {{stack-manage-app}} access to turn on the `alerting:v2:enabled` advanced setting.
```

**Steps**

1. Go to the **Advanced Settings** menu using the navigation menu or the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md).
2. Under **Global settings**, toggle on **alerting:v2:enabled**.
:::

:::{applies-item} serverless:

**Role requirements**

> You must have the `admin` project role to turn on this setting. 

Author: ```suggestion
You must have the `admin` project role to turn on the `alerting:v2:enabled` advanced setting. 
```

**Step**

{{serverless-short}} has no Global Advanced Settings UI, so use Dev Tools to call the global settings API:

```json
POST kbn:/internal/kibana/global_settings
{
  "changes": {
    "alerting:v2:enabled": true
  }
}
```

:::{note}
The `/internal/kibana/global_settings` endpoint is an internal API and might change without notice. There is currently no public equivalent.
:::
:::
::::

### Confirm the UI is accessible [alerting-setup-confirm]

After turning on the setting, verify the {{alerting-v2-system}} is accessible in your space:

> 1. Use the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md) and enter `Alerting v2 preview`.

Author: ```suggestion
1. Use the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md) and enter `Alerting V2 Preview`.
```

2. Select the menu item from the results.

If the menu item doesn't appear immediately, refresh the page and search again. It might take a moment for the UI to reflect the updated setting.

## Turn off the system [alerting-setup-turn-off]

To turn off the {{alerting-v2-system}}, set `alerting:v2:enabled` to `false`.

::::{applies-switch}
:::{applies-item} stack: experimental 9.5+

Go to the **Advanced Settings** page and toggle off **alerting:v2:enabled**.
:::

:::{applies-item} serverless:

Use Dev Tools to call the global settings API:

```json
POST kbn:/internal/kibana/global_settings
{
  "changes": {
    "alerting:v2:enabled": false
  }
}
```
:::
::::

Turning off the setting does not delete any data. {{kib}} retains your rules and action policies as saved objects, and keeps existing documents in `.rule-events` and `.alert-actions`. Turning the setting back on restores the {{alerting-v2-system}} UI.

:::{important}
Turning off `alerting:v2:enabled` hides the {{alerting-v2-system}} UI but does not stop rules and action policies from running. To stop both entirely:

> - **{{stack}}**: Set `xpack.alerting_v2.enabled: false` in [`kibana.yml`](/deploy-manage/deploy/self-managed/configure-kibana.md)

Author: ```suggestion
- **{{stack}}** - Set `xpack.alerting_v2.enabled: false` in [`kibana.yml`](/deploy-manage/deploy/self-managed/configure-kibana.md)
```

> - **{{serverless-short}}**: On {{serverless-short}}, the {{alerting-v2-system}} is managed by Elastic. [Contact Elastic support](https://www.elastic.co/docs/troubleshoot) to turn it off.

Author: ```suggestion
- **{{serverless-short}}** - On {{serverless-short}}, the {{alerting-v2-system}} is managed by Elastic. [Contact Elastic support](https://www.elastic.co/docs/troubleshoot) to turn it off.
```

1. Use the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md) and enter `Alerting v2 preview`.
2. Select the menu item from the results.

If the menu item doesn't appear immediately, refresh the page and search again. It might take a moment for the UI to reflect the updated setting.

## Turn off the system [alerting-setup-turn-off]

To turn off the {{alerting-v2-system}}, set `alerting:v2:enabled` to `false`:

- **{{stack}}:** Go to the same **Advanced Settings** page and toggle off **alerting:v2:enabled**.
- **{{serverless-short}}:** Call the global settings API with `"alerting:v2:enabled": false`.

When the setting is off:

- Rule and action policy execution stops.
- The APIs and UI are hidden.
- Existing rules and action policies are paused.

Turning the setting back on resumes execution. Turning it off does not delete any data. Your rules and action policies remain as {{kib}} saved objects, and existing documents in `.rule-events` and `.alert-actions` are preserved.

## Next steps

After turning on the system, configure role access so your team can use it:

> - **[Configure access](configure-access.md):** Create or update a role with access to the {{alerting-v2-system}} features and the data streams they write to. Users need at minimum **Rules: All** to create rules and `read` index access on `.rule-events` to query rule output in Discover.

Reviewer: double check on the index access required

Author: Updated in [`74d1716` (this PR)](https://github.com/elastic/docs-content/pull/6522/changes/74d1716646b242b681efb2b4156143b9ef5b3a38).
